plugins/csv-parser/dist/types/dataParser.d.ts

```diff
@@ -16,11 +16,14 @@ export interface csvTypes {
     content?: string;
     options?: optionsTypes;
 }
-export type AnimTarget = Anim.AnimTarget | {
-    data: {
-        csv: csvTypes;
+export interface csvTarget {
+    target: {
+        data: {
+            csv: csvTypes;
+        };
     };
-};
+}
+export type AnimTarget = Array<Anim.AnimTarget | csvTarget>;
```

Comment:
Anim.AnimTarget is already the keyframe array. Also, intersection type is not fully solving the problem here.
What if the used wants to set the config beside the csv data:
```javascript
const x: AnimTarget = [
  {
    target: {
      data: {
        csv: {
          url: 'https://raw.githubusercontent.com/vizzuhq/vizzu-lib/master/datasets/olympics.csv'
        }
      },
      config: {
        channels: {
          x: { set: ['blabla'] }
        }
      }
    }
  }
]
``` 
I think here we need a more comlex solution, like:
```
import { MergeDeep } from 'type-fest'

export interface csvTarget {
  target?: {
    data?: {
      csv?: csvTypes
    }
  }
}

export type AnimTarget = MergeDeep<Anim.AnimTarget, Array<csvTarget>>
```
csvTarget props has to be optional in this case.

Reply:
Thanks,  adopted this solution